### ISSUE TITLE
Search with num_workers > 1 and torch 1.5.0 hangs indefinetly

### ISSUE THREAD

**Issue by samuelbroscheit** [uma-pi1/kge#117]
As title says: currently (ax-)search with num_workers > 1 and torch 1.5.0 hangs indefinetly. Due to this setup.py is currently fixed to torch 1.4.0 with commit https://github.com/uma-pi1/kge/commit/ac19d33c08484a23e8475ba707d8f1aa8b9295f6 . Please undo https://github.com/uma-pi1/kge/commit/ac19d33c08484a23e8475ba707d8f1aa8b9295f6 soon as this issue is fixed.

To reproduce install libkge with 

```
git checkout f966612ff4ab9d63c8b69825d39d32dda7eca611
pip install -e .

```
which install torch 1.5.0 and then run the following config. 

```
job.type: search
dataset.name: fb15k-237
train.optimizer: Adagrad
model: complex

search.num_workers: 4
search.type: ax
ax_search:
  num_trials: 20
  parameters:
    - name: train.optimizer_args.lr
      type: range
      bounds: [0.0, 1.0]
```
The hang occurs in Trainer.run_epoch()

for 1vsAll and KvsAll in line:

https://github.com/uma-pi1/kge/blob/ac19d33c08484a23e8475ba707d8f1aa8b9295f6/kge/job/train.py#L300

for Negative Sampling in line:

https://github.com/uma-pi1/kge/blob/ac19d33c08484a23e8475ba707d8f1aa8b9295f6/kge/job/train.py#L315

[edited config]

**Comment by rgemulla** [comment]
Confirmed. For now, if this bug is a problem, one should downgrade to PyTorch 1.4 manually. I reverted the change to master.

The config above is broken. Should be:
```yaml
job.type: search
dataset.name: fb15k-237
train.optimizer: Adagrad
model: complex

search.num_workers: 4
search.type: ax
ax_search:
  num_trials: 20
  parameters:
    - name: train.optimizer_args.lr
      type: range
      bounds: [0.0, 1.0]
```

**Comment by samuelbroscheit** [comment]
Manually downgrading to Pytorch 1.4.0 does not work alone if the other packages gptorch and botorch that depend on Pytorch 1.5.0 are not downgraded as well. 

Another temporary fix to not leave search broken: not allow search.num_workers > 1 for the time beeing.

**Comment by samuelbroscheit** [comment]
We should not leave it broken and let people stumble into this probelm right?

**Comment by rgemulla** [comment]
No, we should fix it. (You may create a temporary branch and point to it from here, if you want.)

**Comment by rgemulla** [comment]
Fixed in 511c6427d8dbd763f807feebbb99aa6902127269
